### ISSUE TITLE
Trace stack is pretty much useless under LuaJIT

### ISSUE THREAD

**Issue by alerque** [sile-typesetter/sile#1013]
I've been using a LuaJIT build for almost everything this week because it is so stinking fast, but one thing I miss is the `-t` argument. LuaJIT doesn't have the `debug` object enabled so it can't find locations of running code.

We should either warn about why it doesn't work or find a way to patch it back when our local debug mode(s) are active.

**Comment by alerque** [comment]
The LuaJIT dilemma is serious. The 2× speed up is *so so* easy to get used to when working with content. When mucking about with anything else (layout, style, special functions, etc.) the lack of debugging info hurts. Swapping out SILE installs or ever switching invocations is more hassle than it should be. Ideally any debug mode or trace flags should just add on the necessary machinery for debugging to work.

Cross linking #982 because it just occurred to me that might help minimize the differences.

**Comment by alerque** [comment]
Um, something else is screwing here. Our stack trace isn't working, but it's not because the `debug` object isn't there at all:

```lua
local foo = function ()
    local trace = debug.traceback("", 2):match("[\n%s]*(.-)$")
    print("my string at "..trace)
end

local bar = function ()
    foo()
end

bar()
```

Works in Lua / LuaJIT / OpenResty.

**Comment by alerque** [comment]
cf. #1069

**Comment by alerque** [comment]
I finally got serious about tackling this ... only to find that it's already fixed. I only ran `git bisect` because I was so miffed and 'curious' at how much time I've wasted waiting around for Lua 5.4 in the last 2 years. The answer seems to be commit cc58824 that made it into v0.14.0, almost certainly as a result of changing `debug.traceback(nil)` to `debug.traceback("")`. I suppose I'm relieved it was relatively recently and I haven't missed out on *that* much render time. I still don't want to calculate how much render time I've wasted.

I'm now going to start looking into the other two LuaJIT related bugs and then into switching some containers and distro packages over to LuaJIT by default.
